resources/js/components/Listing/Listing.vue

```diff
@@ -49,23 +48,115 @@ export default {
     },
 
     computed: {
+        // TODO: Not sure if this is the right place,
+        // the autocomplete also needs this but
+        // we don't want to load everything
+        // directly due the JS size
```

Comment:
I suggest creating another js file with exports to import in the listing and autocomplete for the basic functionality needed everywhere.

If you look at the current situation we also have a similar file for the base of reactivesearch.

If you open the autocomplete you'll see it also downloads a "Container" js file
![image](https://github.com/user-attachments/assets/52611862-6a30-42bb-ac49-db56d893613b)

This also happens when you open a category page:
![image](https://github.com/user-attachments/assets/71c07c1a-2a9a-409b-983a-106ebf25549b)

Vite will automatically recognise and create a separate bundle if it notices multiple other bundles require it.